sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkParserUtils.scala

```diff
@@ -217,14 +225,36 @@ trait SparkParserUtils {
       objectName: Option[String]): Origin = {
     val startOpt = Option(startToken)
     val stopOpt = Option(stopToken)
+
+    // Get the current origin to check for position mapper
+    val currentOrigin = CurrentOrigin.get
+
+    // Don't map positions yet - store them as-is along with the mapper
+    // Position mapping will be applied when creating query context for errors
+    val (startIndex, stopIndex, text, mapper) =
```

Comment:
`startIndex` and `stopIndex` are the same between these two branches, we don't need to put them here, but just `val (text, mapper) =`